src/pkg/packager/common.go

```diff
@@ -353,3 +363,110 @@ func (p *Packager) handleIfPartialPkg() error {
+	err = filepath.Walk(p.tmp.Base, func(path string, info os.FileInfo, err error) error {
+		if !info.IsDir() {
+			filepathMap[path] = false
+		}
+		return nil
+	})
+	if err != nil {
+		return err
+	}
+
+	// Verify the that checksums.txt file matches the aggregated checksum provided
+	actualAggregateChecksum, err := utils.GetSHA256OfFile(p.tmp.Checksums)
+	if err != nil {
+		return fmt.Errorf("unable to get the checksum of the checksums.txt file: %w", err)
+	}
+	if actualAggregateChecksum != p.cfg.Pkg.Metadata.AggregateChecksum {
+		return fmt.Errorf("mismatch on the checksum of the checksums.txt file, the checksums.txt file might have been tampered with")
+	}
+
+	// Check off all the files that we can trust given the checksum and signing checks
+	filepathMap[p.tmp.Checksums] = true
+	filepathMap[p.tmp.ZarfYaml] = true
+	filepathMap[p.tmp.ZarfSig] = true
+
+	// Load the contents of the checksums file
+	checksumsFile, err := os.Open(filepath.Join(p.tmp.Base, "checksums.txt"))
+	if err != nil {
+		return err
+	}
+	defer checksumsFile.Close()
+
+	/* Process every line in the checksums file */
+	scanner := bufio.NewScanner(checksumsFile)
+	scanner.Split(bufio.ScanLines)
+	for scanner.Scan() {
+		// Separate the checksum from the file path
+		strs := strings.Split(scanner.Text(), " ")
+		itemPath := strs[1]
+		expectedShasum := strs[0]
+
+		actualShasum, err := utils.GetSHA256OfFile(filepath.Join(p.tmp.Base, itemPath))
+		if err != nil {
+			return err
+		}
+
+		if expectedShasum != actualShasum {
+			return fmt.Errorf("mismatch on the checksum of the %s file (expected: %s, actual: %s)", itemPath, expectedShasum, actualShasum)
+		}
+
+		filepathMap[filepath.Join(p.tmp.Base, itemPath)] = true
+	}
+
+	for path, processed := range filepathMap {
+		if !processed {
+			return fmt.Errorf("the file %s was present in the Zarf package but not specified in the checksums.txt, the package might have been tampered with", path)
+		}
+	}
+
+	message.Successf("All of the checksums matched!")
+	return nil
+}
+
+func (p *Packager) validatePackageSignature(publicKeyPath string) error {
+
+	// If the insecure flag was provided, ignore the signature validation
+	if config.CommonOptions.Insecure {
+		return nil
+	}
+
+	// Handle situations where there is no signature within the package
+	_, sigCheckErr := os.Stat(p.tmp.ZarfSig)
+	if sigCheckErr != nil {
+		// Nobody was expecting a signature, so we can just return
+		if publicKeyPath == "" {
+			return nil
+		}
+
+		// We were expecting a signature, but there wasn't one..
+		return fmt.Errorf("package is not signed but a key was provided")
+	}
+
+	// Validate the signature of the package
+	if publicKeyPath == "" {
+		return fmt.Errorf("package is signed but no key was provided, using signed packages requires a --key or --insecure flag to continue")
+	}
+
+	// Validate the signature with the key we were provided
+	if err := utils.CosignVerifyBlob(p.tmp.ZarfYaml, p.tmp.ZarfSig, publicKeyPath); err != nil {
```

Comment:
Is there a way to control the Verified OK text color?

Reply:
As we discussed in person, we're going to leave the `Verified OK` from cosign but then make a green descriptive log afterwards to make it clear what has actually been verified successfully